### PR TITLE
ServerConnectionProps  simplify

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Overview
- removed unused `host` variable is unused
- refactor to use lombok generated builder and getter

## Functional Changes
- none

## Manual Testing Performed
- none

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java [hunk 1]
```diff
@@ -191,14 +191,12 @@ private void gameDataChanged() {
   private Optional<ServerConnectionProps> getServerProps(final Component ui) {
     if (System.getProperty(TRIPLEA_SERVER, "false").equals("true")
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java]: As a side note: The TRIPLEA_SERVER and TRIPLEA_CLIENT properties are some temporary properties as well.
AFAIK those are only being used to control the code flow in order to either create a server or a client i.e. host or join a game.
I know the properties are being used when trying to connect to or host a lobby game when we spin up a new instance of the JVM, but I'm not sure if they're used elsewhere.

**Comment by DanVanAtta** [comment]
@prastle please open a dedicated issue for that if you would (and in general, pls favor small focused issues on specific topics compared to piggy-backing on something else)

**Comment by DanVanAtta** [comment]
@prastle remove "-Ptriplea.game.host.console=false"

There are some more CLI changes potentially coming, it's not stable what it will be just yet.

**Comment by prastle** [comment]
@DanVanAtta as I read above this concerns the headless server.  Which has not worked in quite awhile for private hosters.  Since I would assume triplea wants many others to be able to host multiple games like the past this should be part of this?  Sorry if I misunderstood.

**Comment by DanVanAtta** [comment]
> as I read above this concerns the headless server. 

It does, but there are many aspects to it. The code base for headless server is many thousands of lines. This is an update to just a few of those.


> Which has not worked in quite awhile for private hosters. 

This is news, but anecdotal reports like this I'm afraid are not going to help and are just going to demoralize us. Please create a github issue so we can talk about exactly what the problem is there, we can then track any data we collect, we can track any decisions or discussion about good/bad approaches, and we'll be able to ensure it is fixed while the issue is open. Unless you are opening a new github issue, please use a rule of thumb to not say anything is broken (otherwise reference the bug report, if there is no bug report, it works.. I know that is not the reality, but our practice is to work off of bug reports, one guy saying something does not work will not get the problem fixed).

> Since I would assume triplea wants many others to be able to host multiple games like the past this should be part of this? 

This is another topic, probably suitable for forum. The answer involves more than just me and will not get the right visibility here. @prastle please split up topics and do not latch onto recent off-topic threads.  This code update has nothing to do with the last 4 threads. It becomes a problem as we may review this update later, and it'll be a tax to go through and read so much irrelevant information. For many reasons, please put individual questions/topics to new issues or forum threads.

**Comment by prastle** [comment]
@DanVanAtta This has already been discussed in the forum by scousemart @RoiEXLab and myself.

https://forums.triplea-game.org/topic/593/how-to-host-a-bot/8  

AFAIK they were unable to solve the issue.  Sorry if I hijacked.

**Pull request merged**